core-bundle/src/Resources/contao/modules/ModuleLogin.php

```diff
@@ -75,6 +75,11 @@ public function generate()
 				$this->targetPath = $request->query->get('redirect');
 			}
 		}
+		// Use the HTTP referer as a fallback, but only if scheme and host matches with the current request (#5860)
+		elseif ($this->redirectBack && ($referer = $request->headers->get('referer')) && str_starts_with($referer, $request->getSchemeAndHttpHost()))
```

Comment:
```suggestion
		elseif ($this->redirectBack && ($referer = $request->headers->get('referer')) && str_starts_with($referer, $request->getSchemeAndHttpHost().'/'))
```

Reply:
And we should probably normalize the URL to make sure it is safely encoded. E.g. by parsing it with `Nyholm\Psr7\Uri()`.